src/main/java/io/vertx/core/dns/impl/DnsClientImpl.java

```diff
@@ -279,6 +279,14 @@ public DnsClient reverseLookup(String address, Handler<AsyncResult<String>> hand
     return promise.future();
   }
 
+  private long dnsMessageId(int id, String query) {
+    query = query.toLowerCase();
```

Comment:
The trailing dot is from the DNS spec. And we can see it in Netty Codec, [they are adding the trailing dot in the decode domain name function](https://github.com/netty/netty/blob/6150a0e3c5af100e35e5fda3a1d3e3b647bce6a4/codec-dns/src/main/java/io/netty/handler/codec/dns/DnsCodecUtil.java#L114-L119), so we need to add it when it's not present (we see the dot even with `dig` command line).

I thought DNS servers were also insensitive to the case, but apparently not, so I can remove the `toLowerCase()`

Reply:
I see, in this case I think we should in the client itself add a trailing dot to our queries when they are performed instead of having the encoder do it for us, e.g in the `Query` constructor, that would be more clear rather than doing it when computing the message id.